testing/features/forms/step_definitions/input_steps.rb

```diff
@@ -51,3 +51,10 @@
 Then("the text box has a value of {string}") do |text|
   expect(@form.input.value).to eq(text)
 end
+
+Then("I am able to use Inputs of a variety of widths") do
+  @form = Input::FixedWidths.new.tap(&:load)
+  input_types = @form.inputs.map { |input| input["class"] }
+
+  expect(input_types).to be_uniq
```

Comment:
Sorry! One more 🤓 question. Does this `be_uniq` come from the fact that there is an `Array#uniq?` core extension in the `ca_testing` gem?

Reply:
Sorry I figured I may as well explain how this works. And in general metaprogramming.

Ruby heavily uses metaprogramming or programming-on-the-fly. In it you don't know what the code will be until **after** runtime or compilation time (Ruby kinda compiles). As such often we like to do things like this in ruby.

RSpec if loaded (Specifically rspec expectations and rspec matchers), will then scan your codebase for all booleans (Things ending in `?`), and specifically will generate a series of rspec matchers with associated names. The name for all matchers becomes as follows.

- Any boolean named `has_xxx?` creates a matcher called `have_xxx`
- Any boolean named `xxx?` creates a matcher called `be_xxx`

There are also a whole host of other things. RSpec heavily promotes metaprogramming / BDD so it kinda fits hand in glove with step defs.

Reply:
Cheers. Relatively familiar with that pattern. Not 100% sure why we maintain our own core extensions; but good to know that assumption is correct.

Reply:
This is a method that doesn't exist in any library, also these kinds of methods (The ones written), are good extensions to have, so they're added to the ca_testing gem (Alongside patches and driver methods).